layers/+lang/scala/packages.el

```diff
@@ -191,6 +192,9 @@
       (when (configuration-layer/package-usedp 'expand-region)
         (require 'ensime-expand-region nil 'noerror)))))
 
+(defun scala/post-init-flycheck ()
+  (add-hook 'scala-mode-hook 'flycheck-mode))
```

Comment:
LOL I took as reference a random lang layer (php) and that's why I used that 😆 
I just updated the PR with your suggestion (also fixed php)